DKP-222: Add release notes for 4.10.0 (#15015)
docker/docker.github.io: desktop/release-notes.md

> [RPM](https://desktop.docker.com/linux/main/amd64/docker-desktop-4.10.0-x86_64.rpm?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-linux-amd64){: .button .primary-btn }
> [Arch (experimental)](https://desktop.docker.com/linux/main/amd64/docker-desktop-4.10.0-x86_64.pkg.tar.zst?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-linux-amd64){: .button .primary-btn }

### Upgrades
- [Docker Engine to v20.10.17](https://docs.docker.com/engine/release-notes/#201017)
- [Docker Compose v2.6.1](https://github.com/docker/compose/releases/tag/v2.6.1)
- [Kubernetes v1.41.1](https://github.com/kubernetes/kubernetes/releases/tag/v1.41.1)
- [cri-dockerd to v0.2.1](https://github.com/Mirantis/cri-dockerd/releases/tag/v0.2.1)
- [CNI plugins to v1.1.1](https://github.com/containernetworking/releases/tag/v1.1.1)
- [containerd to v1.6.6](https://github.com/containerd/containerd/releases/tag/v1.6.6)
- [runc to v1.1.2](https://github.com/opencontainers/runc/releases/tag/v1.1.2)
- [Go 1.18.3](https://github.com/golang/go/releases/tag/go1.18.3)

> ### New

Reviewer: I think there's too much in New here. We normally reserve New for actual new features. Most of these should be in the Minor Changes section.

Reviewer 2: `New` usually matches the release highlights, so maybe we can wait until we settle on them and revisit here?

Author: I've move `New` content to `Bug fixes and minor changes` in the meantime, as well as convert to draft to avoid wrong merge.

- Expanded / collapsed state of the Compose apps is now persisted.
- External URLs can refer to extension details in Docker Desktop Marketplace using links like `docker-desktop://extensions/marketplace?extensionId=docker/logs-explorer-extension`.
- Containers overview: added bulk actions for start/pause/stop selected containers.
- Containers overview: pause/restart actions for compose.
- Implemented changes on the logs search functionality and added the possibility of clearing the terminal.
- `docker extension` CLI commands are available with Docker Desktop by default.

> - Disabled Homepage while working on a new design. You can provide [feedback here](https://docs.google.com/forms/d/e/1FAIpQLSfYueBkJHdgxqsWcQn4VzBn2swu4u_rMQRIMa8LExYb_72mmQ/viewform?entry.1237514594=4.10).

Reviewer: I think we have a different section for removed features.

Also this should say "Removed" not "Disabled".

Author: @stephen-turner `Deprecation` section maybe ?

Reviewer 2: Deprecation sounds a little different, as it’s a step before actually removing something.
However looking at the history, it looks like we’ve used that in the past. 🤷

Reviewer: It's not Deprecation, we can just call it Removal.

> [Mac with Intel chip](https://desktop.docker.com/mac/main/amd64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-amd64){: .button .primary-btn }
> [Mac with Apple chip](https://desktop.docker.com/mac/main/arm64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-arm64){: .button .primary-btn }
> [DEB](https://desktop.docker.com/linux/main/amd64/docker-desktop-4.10.0-amd64.deb?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-linux-amd64){: .button .primary-btn }
> [RPM](https://desktop.docker.com/linux/main/amd64/docker-desktop-4.10.0-x86_64.rpm?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-linux-amd64){: .button .primary-btn }
> [Arch (experimental)](https://desktop.docker.com/linux/main/amd64/docker-desktop-4.10.0-x86_64.pkg.tar.zst?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-linux-amd64){: .button .primary-btn }

### Upgrades
- [Docker Engine to v20.10.17](https://docs.docker.com/engine/release-notes/#201017)
- [Docker Compose v2.6.1](https://github.com/docker/compose/releases/tag/v2.6.1)

> - [Kubernetes v1.41.1](https://github.com/kubernetes/kubernetes/releases/tag/v1.41.1)

Reviewer 3: I don't think this is right - latest Kubernetes version is v1.24.x (link is actually broken here)

### New

- TODO: Waiting for highlights.

### Bug fixes and minor changes

#### For all platforms

- Containers tab: Bulk actions for start/pause/stop selected containers.
- Containers tab: Pause/restart actions for compose projects.
- External URLs can refer to extension details in Docker Desktop Marketplace using links like `docker-desktop://extensions/marketplace?extensionId=docker/logs-explorer-extension`.
- Implemented changes on the logs search functionality and added the possibility of clearing the terminal.
- Expanded / collapsed state of the Compose apps is now persisted.
- `docker extension` CLI commands are available with Docker Desktop by default.
- Extensions: make screenshots displayed in the marketplace bigger.
- Fixed a bug where a Docker extension fails to load if its backend container(s) are stopped. Fixes [docker/extensions-sdk#16](https://github.com/docker/extensions-sdk/issues/162).
- Fixed a bug where license agreement is not showing up, thus silently blocking Docker Desktop startup.
- Fixed the displayed image and tag for unpublished extensions to actually display the ones from the installed unpublished extension.

> - Fixed the duplicate footer on the Support screen.

Reviewer 4: From the Environments team, I'd like to add a line for the work done in https://github.com/docker/dev-envs/pull/186, which addresses https://github.com/docker/dev-environments/issues/16

```suggestion
- Fixed the duplicate footer on the Support screen.
- Dev Environments can be created from a subdirectory in a GitHub repository
```

#### For Mac

- Fixed a bug with location of bash completion files on macOS. Fixes [docker/for-mac#6343](https://github.com/docker/for-mac/issues/6343).
- Fixed a bug where Docker would not start if the username was longer than 25 characters. Fixes [docker/for-mac#6122](https://github.com/docker/for-mac/issues/6122).
- Fixed a bug when Docker Desktop was not starting due to invalid system proxy configuration. Fixes some issues reported in [docker/for-mac#6289](https://github.com/docker/for-mac/issues/6289).

#### For Windows

- Fixed [docker/for-win#12738](https://github.com/docker/for-win/issues/12738).

> - Fixed a bug where the dashboard was empty. Fixes [docker/for-win#12759](https://github.com/docker/for-win/issues/12759).

Author: You're right, it was cherry picked 

https://github.com/docker/pinata/pull/18188

Good catch, thanks!

> [RPM](https://desktop.docker.com/linux/main/amd64/docker-desktop-4.10.0-x86_64.rpm?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-linux-amd64){: .button .primary-btn }
> [Arch (experimental)](https://desktop.docker.com/linux/main/amd64/docker-desktop-4.10.0-x86_64.pkg.tar.zst?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-linux-amd64){: .button .primary-btn }

### Security

Update Go runtime to 1.18.3 to address:

- [CVE-2022-30634](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-30634)
- [CVE-2022-30629](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-30629)
- [CVE-2022-30580](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-30580)
- [CVE-2022-29804](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-29804)

### Upgrades
- [Docker Compose v2.6.1](https://github.com/docker/compose/releases/tag/v2.6.1)
- [Kubernetes v1.41.1](https://github.com/kubernetes/kubernetes/releases/tag/v1.41.1)
- [cri-dockerd to v0.2.1](https://github.com/Mirantis/cri-dockerd/releases/tag/v0.2.1)
- [CNI plugins to v1.1.1](https://github.com/containernetworking/releases/tag/v1.1.1)

> - [Docker Engine to v20.10.17](https://docs.docker.com/engine/release-notes/#201017)

Reviewer 2: Maybe this one should be listed first in the upgrades section.

- External URLs can refer to extension details in Docker Desktop Marketplace using links like `docker-desktop://extensions/marketplace?extensionId=docker/logs-explorer-extension`.
- Containers overview: added bulk actions for start/pause/stop selected containers.
- Containers overview: pause/restart actions for compose.
- Implemented changes on the logs search functionality and added the possibility of clearing the terminal.
- `docker extension` CLI commands are available with Docker Desktop by default.
- Disabled Homepage while working on a new design. You can provide [feedback here](https://docs.google.com/forms/d/e/1FAIpQLSfYueBkJHdgxqsWcQn4VzBn2swu4u_rMQRIMa8LExYb_72mmQ/viewform?entry.1237514594=4.10).

### Bug fixes and minor changes

#### For all platforms

- Fixed a bug where a Docker extension fails to load if its backend container(s) are stopped. Fixes [docker/extensions-sdk#16](https://github.com/docker/extensions-sdk/issues/162).
- Fixed a bug where license agreement is not showing up, thus silently blocking Docker Desktop startup.
- Fixed the displayed image and tag for unpublished extensions to actually display the ones from the installed unpublished extension.

> - Fixed the duplicate footer on the Support screen.

Reviewer 2: Don’t we have any for-win/for-mac links for these?

Author: Dunno, I'll ping author accordingly

> [RPM](https://desktop.docker.com/linux/main/amd64/docker-desktop-4.10.0-x86_64.rpm?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-linux-amd64){: .button .primary-btn }
> [Arch (experimental)](https://desktop.docker.com/linux/main/amd64/docker-desktop-4.10.0-x86_64.pkg.tar.zst?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-linux-amd64){: .button .primary-btn }

### Upgrades
- [Docker Engine to v20.10.17](https://docs.docker.com/engine/release-notes/#201017)
- [Docker Compose v2.6.1](https://github.com/docker/compose/releases/tag/v2.6.1)
- [Kubernetes v1.24.1](https://github.com/kubernetes/kubernetes/releases/tag/v1.24.1)
- [cri-dockerd to v0.2.1](https://github.com/Mirantis/cri-dockerd/releases/tag/v0.2.1)
- [CNI plugins to v1.1.1](https://github.com/containernetworking/releases/tag/v1.1.1)
- [containerd to v1.6.6](https://github.com/containerd/containerd/releases/tag/v1.6.6)
- [runc to v1.1.2](https://github.com/opencontainers/runc/releases/tag/v1.1.2)
- [Go 1.18.3](https://github.com/golang/go/releases/tag/go1.18.3)

### New

> - User can now add environment variables when running an image right from the Docker Dashboard.

Reviewer 5: ```suggestion
- You can now add environment variables before running an image in Docker Desktop.
```

- Added features to make it easier to work with containers logs such as regex search and the ability to clear container logs while the container is still running.
- Addressed feedback on new containers tables, added ports and separated container and image names.
- Two new extensions (Ddosify and Lacework) added to the Extensions Marketplace.

### Bug fixes and minor changes

#### For all platforms

- Containers tab: Bulk actions for start/pause/stop selected containers.
- Containers tab: Pause/restart actions for compose projects.
- Containers tab: Adds icons as well as the exposed port(s) / exit code.
- External URLs can refer to extension details in Docker Desktop Marketplace using links like `docker-desktop://extensions/marketplace?extensionId=docker/logs-explorer-extension`.

> - Implemented changes on the logs search functionality and added the possibility of clearing the terminal.

Reviewer 5: Yes please!

- Expanded / collapsed state of the Compose apps is now persisted.
- `docker extension` CLI commands are available with Docker Desktop by default.

> - Extensions: make screenshots displayed in the marketplace bigger.

Reviewer 5: ```suggestion
- Increased the size of the screenshots displayed in the Extension marketplace.
```

- Fixed a bug where a Docker extension fails to load if its backend container(s) are stopped. Fixes [docker/extensions-sdk#16](https://github.com/docker/extensions-sdk/issues/162).

> - Fixed a bug where image search field was cleared without reason. Fixes [docker/for-win#12738](https://github.com/docker/for-win/issues/12738).

Reviewer 5: ```suggestion
- Fixed a bug where the image search field is cleared without a reason. Fixes [docker/for-win#12738](https://github.com/docker/for-win/issues/12738).
```

> - Fixed a bug where license agreement is not showing up, thus silently blocking Docker Desktop startup.

Reviewer 5: ```suggestion
- Fixed a bug where the license agreement does not display and silently blocks Docker Desktop startup.
```

- Fixed the displayed image and tag for unpublished extensions to actually display the ones from the installed unpublished extension.
- Fixed the duplicate footer on the Support screen.

> - Dev Environments can be created from a subdirectory in a GitHub repository.

Author: Don't know, it was not in release highlights so I would stay it is fine here ?

> - Occasionally the Docker engine will restart during a `docker system prune`. This is a [known issue](https://github.com/moby/buildkit/pull/2177) in the version of buildkit used in the current engine and will be fixed in future releases.

Reviewer 5: This needs to be in a 'Known issue' section and with a 'For all platforms' subheading. Similar to the screenshot 

![image](https://user-images.githubusercontent.com/102604716/176192308-8c06740c-8dea-4bd4-803a-3bb4e2404f98.png)

> - Removed the error message if the tips of the day cannot be loaded when using Docker Desktop offline. Fixes [docker/for-mac#6366](https://github.com/docker/for-mac/issues/6366).

Reviewer 5: Think this should go in the 'removal' section

#### For Mac

- Fixed a bug with location of bash completion files on macOS. Fixes [docker/for-mac#6343](https://github.com/docker/for-mac/issues/6343).

> - Fixed a bug where Docker would not start if the username was longer than 25 characters. Fixes [docker/for-mac#6122](https://github.com/docker/for-mac/issues/6122).

Reviewer 5: ```suggestion
- Fixed a bug where Docker Desktop does not start if the username is longer than 25 characters. Fixes [docker/for-mac#6122](https://github.com/docker/for-mac/issues/6122).
```

> - Fixed a bug when Docker Desktop was not starting due to invalid system proxy configuration. Fixes some issues reported in [docker/for-mac#6289](https://github.com/docker/for-mac/issues/6289).

Reviewer 5: ```suggestion
- Fixed a bug where Docker Desktop was not starting due to invalid system proxy configuration. Fixes some issues reported in [docker/for-mac#6289](https://github.com/docker/for-mac/issues/6289).
```

#### For Windows

- Fixed high CPU usage on Hyper-V [docker/for-win#12780](https://github.com/docker/for-win/issues/12780).

#### For Linux

- Fixed a bug when settings cannot be applied more than once.
- Fixed Compose version displayed in the `About` screen.

> ### Removal

Reviewer 5: ```suggestion
### Removed
```

> 

Reviewer 5: Can we add a 'For all platforms' heading here?

> [Mac with Intel chip](https://desktop.docker.com/mac/main/amd64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-amd64){: .button .primary-btn }
> [Mac with Apple chip](https://desktop.docker.com/mac/main/arm64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-arm64){: .button .primary-btn }
> [DEB](https://desktop.docker.com/linux/main/amd64/docker-desktop-4.10.0-amd64.deb?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-linux-amd64){: .button .primary-btn }
> [RPM](https://desktop.docker.com/linux/main/amd64/docker-desktop-4.10.0-x86_64.rpm?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-linux-amd64){: .button .primary-btn }
> [Arch (experimental)](https://desktop.docker.com/linux/main/amd64/docker-desktop-4.10.0-x86_64.pkg.tar.zst?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-linux-amd64){: .button .primary-btn }

### New

- You can now add environment variables before running an image in Docker Desktop.

> - Added features to make it easier to work with a container's logs, such as regex search and the ability to clear container logs while the container is still running.

Reviewer: ```suggestion
- Added features to make it easier to work with a container's logs, such as regular expression search and the ability to clear container logs while the container is still running.
```

- Implemented feedback on the containers table. Added ports and separated container and image names.
- Added two new extensions, Ddosify and Lacework, to the Extensions Marketplace.

### Upgrades

> - [Docker Engine to v20.10.17](https://docs.docker.com/engine/release-notes/#201017)

Reviewer: ```suggestion
- [Docker Engine 20.10.17](https://docs.docker.com/engine/release-notes/#201017)
```

- [Docker Compose v2.6.1](https://github.com/docker/compose/releases/tag/v2.6.1)
- [Kubernetes v1.24.1](https://github.com/kubernetes/kubernetes/releases/tag/v1.24.1)
- [cri-dockerd to v0.2.1](https://github.com/Mirantis/cri-dockerd/releases/tag/v0.2.1)
- [CNI plugins to v1.1.1](https://github.com/containernetworking/releases/tag/v1.1.1)
- [containerd to v1.6.6](https://github.com/containerd/containerd/releases/tag/v1.6.6)
- [runc to v1.1.2](https://github.com/opencontainers/runc/releases/tag/v1.1.2)
- [Go 1.18.3](https://github.com/golang/go/releases/tag/go1.18.3)

### Bug fixes and minor changes

#### For all platforms

> - Added additional bulk actions for starting/pausing/stopping select containers in the **Containers** tab.

Reviewer: ```suggestion
- Added additional bulk actions for starting/pausing/stopping selected containers in the **Containers** tab.
```

- Fixed a bug where the image search field is cleared without a reason. Fixes [docker/for-win#12738](https://github.com/docker/for-win/issues/12738).
- Fixed a bug where the license agreement does not display and silently blocks Docker Desktop startup.
- Fixed the displayed image and tag for unpublished extensions to actually display the ones from the installed unpublished extension.
- Fixed the duplicate footer on the Support screen.
- Dev Environments can be created from a subdirectory in a GitHub repository.

#### For Mac

- Fixed a bug with location of bash completion files on macOS. Fixes [docker/for-mac#6343](https://github.com/docker/for-mac/issues/6343).
- Fixed a bug where Docker Desktop does not start if the username is longer than 25 characters. Fixes [docker/for-mac#6122](https://github.com/docker/for-mac/issues/6122).
- Fixed a bug where Docker Desktop was not starting due to invalid system proxy configuration. Fixes some issues reported in [docker/for-mac#6289](https://github.com/docker/for-mac/issues/6289).

> - Fixed a bug where a startup race occurred when the experimental virtualization framework is enabled.

Reviewer: This is the cause: what was the symptom from the user's pov?

Author: I would assume Docker Desktop not starting, @djs55 is that correct ?

Reviewer 6: Symptom was failing to startup (underneath the Swift code started the engine etc, then opened the dashboard, then exited, then everything else would either exit or get stuck depending on timing).

Maybe omit the mention of the race?
```
Fixed a bug where Docker Desktop failed to start when the experimental virtualization.framework is enabled.
```

> - Fixed a bug where tray icon was still displayed after uninstalling Docker Desktop.

Reviewer 5: ```suggestion
- Fixed a bug where the tray icon still displayed after uninstalling Docker Desktop.
```

#### For Windows

- Fixed a bug which caused high CPU usage on Hyper-V. Fixes [docker/for-win#12780](https://github.com/docker/for-win/issues/12780).
- Fixed a bug where Docker Desktop for Windows would fail to start. Fixes [docker/for-win#12784](https://github.com/docker/for-win/issues/12784).

#### For Linux

- Fixed a bug when settings cannot be applied more than once.
- Fixed Compose version displayed in the `About` screen.

> ### Removed

Reviewer: I think the "Removed" section should go higher up, below "New" but above "Upgrades".

> #### For all platforms

Reviewer: I think we can remove the header "for all platforms" if all the things in the section apply to all platforms.

> - Removed the error message if the tips of the day cannot be loaded when using Docker Desktop offline. Fixes [docker/for-mac#6366](https://github.com/docker/for-mac/issues/6366).

Reviewer: This section is for removed functionality. This bullet point is just a bug fix, and should move into the bug fixes section.

- Removed Homepage while working on a new design. You can provide [feedback here](https://docs.google.com/forms/d/e/1FAIpQLSfYueBkJHdgxqsWcQn4VzBn2swu4u_rMQRIMa8LExYb_72mmQ/viewform?entry.1237514594=4.10).

### Known Issues

> #### For all platforms

Reviewer: Remove "For all platforms" here too.
